cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/kafka/LoggingEventSerializer.java

```diff
@@ -77,6 +78,36 @@ public ILoggingEvent fromBytes(ByteBuffer buffer) throws IOException {
   }
 
   /**
+   * Decodes the timestamp of a {@link ILoggingEvent} encoded in the given {@link ByteBuffer} with the schema
+   * returned by the {@link #getAvroSchema()} method.
+   *
+   * @param buffer the buffer to decoder
+   * @return the event timestamp
+   * @throws IOException if fail to decode
+   */
+  public long decodeEventTimestamp(ByteBuffer buffer) throws IOException {
```

Comment:
It will be used in @mao PR #7946 in order to scan Kafka and decode event time stamp in quick manner instead of decoding the whole event. From local micro benchmark, it's about 150 times faster.

The reason I have it throwing IOException is because it is the exception thrown in general when decode fail. We could throw a different exception (e.g. NotFoundException) for not able to find the timestamp field in the scheme (it will only happen if the scheme changed, but the unit test will fail anyway, hence it can't be changed without notice) and from the caller perspective, it should never need to catch the NotFoundException. However, the caller does need to catch and handle the IOException in general to skip bad data, as we can't guard against someone publishing malformed data to a Kafka topic.